src/block-tracker-inspector.ts

```diff
@@ -13,37 +16,64 @@ interface BlockTrackerInspectorMiddlewareOptions {
   blockTracker: PollingBlockTracker;
 }
 
+function hasProperty<Property extends string>(
+  object: unknown,
+  property: Property,
+): object is Record<Property, unknown> {
```

Comment:
This is similar to the one in our utils library: https://github.com/MetaMask/utils/blob/main/src/misc.ts#L93
Except here I used a type predicate to assert that the object had this key. The type narrowing in `getResultBlockNumber` doesn't work without it.

Reply:
I like the alternative suggestion, as it's a bit more semantic and descriptive for any users making their way to the types.

Reply:
Done here: 5b5f61854ca3af20b2e44da1cc4cde947506a61e

I realized that the constraint on `Property` was too restrictive in that it prevented `number` and `symbol` properties, so I moved the valid property types to a separate type and referenced that in both places.

Reply:
Hmm. It seems that when using this function, `object` would become `never` when the function returns false. I don't think you need to redeclare the type of the object entirely — I think you can use `&` to extend its definition. What about this?

```
type ValidPropertyType = string | number | symbol;

function hasProperty2<Object, Property extends ValidPropertyType>(
  object: Object,
  name: Property,
): object is Object & Record<Property, unknown> {
  return Object.hasOwnProperty.call(object, name);
}
```

See playground: https://www.typescriptlang.org/play?ts=4.9.4&ssl=1&ssc=1&pln=2&pc=1#code/C4TwDgpgBAaghgGwJYBMAKAnA9pDoAq40AvFAM7AZIB2A5lAD5TUCuAtgEYQaPkidYEAbgBQIgGYtqAY2BIs1KAAs4ZTDm6gAjAB51uUFAgAPYBGooysRKn2aQhSAD4AFCKhQsHAFYRZALigpAGtqLAB3agAad2Y4NghAuzwQGIBKQK9fWSgkKwAlPywMFD1sAxBeeGR0cvtHCCig6lCI6icoAG9YjAhgFgxFAHkfP2AAOhUyIcjk0HHpRAQXLLGm6niINNEAXzFJGTkFZVU5kAAmHRHs4Cazo1NzS2sas4bXWNWAqGu12I2Ekk6il0plRjk8j9wcAoAAyKCFaTFUpnJohMKRDrdDy9fqDKE3SaqGbUM4LJYraHrTbbER7ERI6gUKDGKCkbFQcRYLCBABEHDgGF5MQ8AoAXnyAI4sYy8un7KSyeSKLlYFxpLqxJDiKAuKZnLQuYxNXniJBisUcFgW3lpDUcjzGWI7IwIMjQB0s8bi51iDza3X64Ggc5Gk1mi1Wm12zUeR3O13u2Nx4zeuBi307IA

Reply:
Good catch, you're right, an intersection type is what was needed here.

Reply:
Oh I see, that was the name of the generic parameter, I missed that. That works.

Reply:
Yeah feel free to name it whatever's less confusing :)

Reply:
Done here: https://github.com/MetaMask/eth-json-rpc-middleware/pull/183/commits/0c3afb7f0e981e6f62d7d5b27e93051bee7e2089